Celigo.ServiceManager.NetSuite.REST/src/RestletClientFactory.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+
+namespace Celigo.ServiceManager.NetSuite.REST
+{
+    public interface IRestletClientFactory
+    {
+        RestletClient CreateClient(string restletName);
+    }
+
+    public class RestletClientFactory : IRestletClientFactory
+    {
+        private readonly Dictionary<string, RestletConfig> _registry;
+        private readonly IHttpClientFactory _clientFactory;
+        private readonly IOptions<RestClientOptions> _restClientOptions;
+
+        public RestletClientFactory(IHttpClientFactory clientFactory, IOptions<RestletConfigOptions> configOptions, IOptions<RestClientOptions> options)
+        {
+            _clientFactory = clientFactory;
+
+            var restlets = configOptions.Value?.Restlets ?? throw new InvalidOperationException("At least one RESTlet must be configured.");
+
+            if (restlets.Any(r => r.Deploy == null || r.Script == null || r.RestletName == null))
+            {
+                throw new InvalidOperationException("At least one RESTlet has not been properly configured.");
+            }
+            
+            _registry = configOptions.Value.Restlets.ToDictionary(c => c.RestletName);
+            _restClientOptions = options;
+        }
+
+        public RestletClient CreateClient(string restletName)

```

Comment:
Instead of passing name, can we pass an enum ?
I think enums will be less error prone.

Reply:
Dictionary can be of type <enumType, RestletConfig>

Reply:
If you think of the cost of adding a generic interface, I think this would be too much trouble for too little gain. You could just define constants for RESTlet names. 
And by "cost", I meant the trouble of dealing with the type parameters everywhere you need it. Given that it only helps you minimize errors in typos (if you didn't use the constant).

Reply:
+1 for constants, and by using enum I meant using constants that are in the form of enum.